libs/swig/io.i

```diff
@@ -94,6 +94,10 @@
   }
 }
 
+%typemap(directorin) (const char *data, int size) {
+    $input = PyBytes_FromStringAndSize($1, $2);
```

Comment:
That means that we go from two parameters to simply one? That would break the API. The conversion from bytes and length seems to be appropriate although I don't know exactly what `PyBytes_FromStringAndSize` does. Is the length of the bytes objects the same as `size`?

Please note that (data, size) could also be  a binary blob, it must not be interpreted as string. I know that the current approach is also wrong but I am not sure if the fix solves all the issues.

Reply:
> That means that we go from two parameters to simply one? That would break the API.

Yes. Maybe we can somehow keep `size` (which is now equivalent to `len(data)` in Python, thus redundant), but changing the type of `data` from `str` to `bytes` breaks the (Python) API anyway.

> I don't know exactly what `PyBytes_FromStringAndSize` does. Is the length of the bytes objects the same as `size`?

Yes.

> Please note that (data, size) could also be a binary blob, it must not be interpreted as string.

This is exactly why we need `PyBytes_FromStringAndSize`. Previously, data was not only interpreted as string, but **unicode** string.

Reply:
Previously `len(data) != size`, because some unicode characters are longer than one byte.

Reply:
I agree that this is not trivial. So actually keeping only the bytes array and omitting the size attribute is the best option.

Reply:
Why does it actually not change `io.py` to reflect the new signature?

Reply:
I don't know. Apparently changing io.py is not needed because of variable arguments.

BTW, I changed wrappers manually, because I have a different SWIG version and it generates completely different wrappers. But I ran fdsnws tests and there were no errors.